engine/src/main/java/org/terasology/engine/ComponentSystemManager.java

```diff
@@ -85,6 +85,14 @@ public void loadSystems(ModuleEnvironment environment, NetworkMode netMode) {
             RegisterSystem registerInfo = type.getAnnotation(RegisterSystem.class);
             if (registerInfo.value().isValidFor(netMode, isHeadless)) {
                 systemsByModule.put(moduleId, type);
+                if(registerInfo.requiresOptional().length != 0) {
```

Comment:
Hi @msteiger , thank you for reviewing !!! 
I've added a method to decide wether to insert the system, could you please help me review it again ? What it does : 

- The method `ifRegister(RegisterSystem registerSystem,ModuleEnvironment enviromnent)` get a list of `ModuleName` that this system has option dependency on, using `RegisterSystem.requiresOptional()`
- If this list is length 0, so this system hasn't optional dependency, it should be registered(true)
- If there is any `ModuleName` aren't present, then it shouldn't be registered, return false
- If all these `ModuleName` are present in environment, then return true

I still have some questions :

- Should I change the name of this method ?
- Should I change the position of this method in class ?